### ISSUE TITLE
[diagnostic] typeof operator should warn when comparing to invalid string literals 

### ISSUE THREAD

**Issue by CoderMuffin** [quick-lint/quick-lint-js#1213]
When performing checks like `typeof x == "string"`, it is perfectly possible to misspell it as something like `typeof x == "strng"` (missing `i`). Maybe there should be a warning for this? I would be happy to implement this if approved :)

**Comment by strager** [comment]
This is a great idea! I'll put 12$ on it.

**Comment by singalhimanshu** [comment]
@milanjijo are you working on this task?

**Comment by singalhimanshu** [comment]
@strager Is there any way to currently check a string literal against this list ("undefined", "object", "boolean", "number", "bigint", "string", "symbol" or "function") in the code?

**Comment by CoderMuffin** [comment]
@singalhimanshu yes, you can compare the string_view of the Expression::Literal to each element of the array. I authored a similar diagnostic, the detection code for which can be seen here: [https://github.com/quick-lint/quick-lint-js/blob/d09a1a72675fd1673dac5793b099c6169698829d/src/quick-lint-js/fe/parse.cpp#L524](src/quick-lint-js/fe/parse.cpp:524)

**Comment by dubloom** [comment]
Is anyone still working on this issue ? Otherwise I would be interested to tackle it as i already have a working piece of code. Thank you !
